doop-image-checker/main.go

```diff
@@ -101,15 +103,46 @@ func handleHeaders(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	//make request to Keppel's Registry API (we deliberately make a GET here in
-	//order to explicitly update the image's last_pulled_at timestamp);
-	//unfortunately we need to engage in some trickery to extract the
-	//X-Keppel-Vulnerability-Status header
 	var hc headerCapturer
-	_, err = remote.Image(ref, remote.WithTransport(&hc))
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+	// if a response config file is provided always use that
+	if len(os.Args) == 3 {
+		byte, err := os.ReadFile(os.Args[2])
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		config := make(map[string]map[string]string)
+		err = yaml.Unmarshal(byte, config)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
```

Comment:
This reads the configuration file on every (uncached) request. The classic criticism is that this is inefficient, which admittedly is not that big of an issue for a testing-only code path. My bigger issue is that this makes for poor separation of concerns. The code structure will be clearer when `os.Args` is only dealt with in `main` to parse the config file once, and then the parsed data is provided to the request handler.